### PR TITLE
Add RBAC implementation for API endpoints which were missing it

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds RBAC implementation for API endpoints which were missing it (endpoints which were added recently where we got sloppy and didn't implement when we were implementing that feature - we should be careful this doesn't repeat again in the future):

- [x] Traces (get all, get one, add RBAC tests for API endpoints)
- [ ] Webhooks (get one, get all, add RBAC tests for API endpoints)
- [x] Timers (get all, get one, add RBAC tests for API endpoints)
- [x] API keys (get one, create, add RBAC tests for API endpoints)

## TODO

- [ ] Docs

### CODE REVIEW

File: st2common/st2common/openapi.yaml [hunk 1]
```diff
@@ -3348,7 +3360,13 @@ paths:
   /api/v1/webhooks:
     get:
       operationId: st2api.controllers.v1.webhooks:webhooks_controller.get_all
+      x-permissions: {{ PERMISSION_TYPE.WEBHOOK_LIST }}
       description: Returns a list of all webhooks.
+      x-parameters:
+        - name: user
+          in: context
+          x-as: requester_user
+          description: User performing the operation
```
Comment (enykeev) [inline st2common/st2common/openapi.yaml]: I don't see why you need both. `x-permissions` should be more than enough here.

File: st2common/st2common/rbac/resolvers.py [hunk 2]
```diff
@@ -769,58 +771,136 @@ def user_has_permission(self, user_db, permission_type):
 
     def user_has_resource_api_permission(self, user_db, resource_api, permission_type):
         assert permission_type in [PermissionType.API_KEY_CREATE]
-        # TODO:
-        return True
+        return self._user_has_global_permission(user_db=user_db, permission_type=permission_type)
```
Comment (enykeev) [inline st2common/st2common/rbac/resolvers.py]: Can't find where we using this one.

Comment (Kami) [inline st2common/st2common/rbac/resolvers.py]: It's not being used by WebhooksController.get_all method - https://github.com/StackStorm/st2/pull/3351/commits/aef02a481fb29cfbc45e216eeac3a77782147454#diff-91ccd907e72f23cb1463520e892c92e6R98.

File: st2common/st2common/openapi.yaml (hunk 1 above)
Comment (Kami) [inline st2common/st2common/openapi.yaml]: Will try it out and verify it works without this parameter.

File: st2common/st2common/rbac/resolvers.py (hunk 2 above)
Comment (enykeev) [inline st2common/st2common/rbac/resolvers.py]: you've answered on the wrong one

Comment (enykeev) [inline st2common/st2common/rbac/resolvers.py]: I also found what's going on in this one. You can ignore the thread.

File: st2common/st2common/openapi.yaml (hunk 1 above)
Comment (Kami) [inline st2common/st2common/openapi.yaml]: Good catch, it's indeed not needed - 07e06d4.

File: st2api/tests/unit/controllers/v1/test_rbac_for_supported_endpoints.py [hunk 3]
```diff
@@ -13,13 +13,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from collections import OrderedDict
 import httplib
+from collections import OrderedDict
```
Comment (enykeev) [inline st2api/tests/unit/controllers/v1/test_rbac_for_supported_endpoints.py]: I've never been able to understand the logic behind ordering the imports

**Pull request merged**